### PR TITLE
support ProduceReqV7 encode and ProduceRespV7 decode

### PR DESCRIPTION
**Pull request by pcHwang2**

### CODE REVIEW

File: codec/const.go [hunk 1]
```diff
@@ -56,4 +56,9 @@ const (
 	LenThrottleTime                       = 4
 	LenTime                               = 8
 	LenTopicAuthOperation                 = 4
+	LenApiKey                             = 2
```
Comment (Shoothzj) [inline codec/const.go:59]: please sort by alphabet?

File: codec/util_kafka_int16.go [hunk 2]
```diff
@@ -68,3 +68,11 @@ func putProducerEpoch(bytes []byte, idx int, errorCode int16) int {
 func readProducerEpoch(bytes []byte, idx int) (int16, int) {
 	return readInt16(bytes, idx)
 }
+
+func putRequiredAcks(bytes []byte, idx int, x int16) int {
```
Comment (Shoothzj) [inline codec/util_kafka_int16.go:72]: please sort by alphabet

File: codec/const.go (hunk 1 above)
Comment (pcHwang2) [reply codec/const.go]: done

File: codec/produce_req_test.go [hunk 3]
```diff
@@ -83,3 +83,19 @@ func TestDecodeProduceReqV8(t *testing.T) {
 	assert.Equal(t, []byte("key"), record.Key)
 	assert.Equal(t, "value", string(record.Value))
 }
+
+func TestDecodeAndCodeReqV7(t *testing.T) {
```
Comment (Shoothzj) [inline codec/produce_req_test.go:87]: let's add a simple codec case. We may have arguments on codec function, like contains apiVersion or not.

**Pull request merged**